### PR TITLE
test(v2): fixed presubmit v2 go test

### PR DESCRIPTION
**Pull request by capri-xiyue**
**Description of your changes:**
fixed presubmit v2 go test

**Checklist:**
- [ ] The title for your pull request (PR) should follow our title convention. [Learn more about the pull request title convention used in this repository](https://github.com/kubeflow/pipelines/blob/master/CONTRIBUTING.md#pull-request-title-convention).

### CODE REVIEW

File: v2/test/presubmit-v2-go-test.sh [hunk 1]
```diff
@@ -35,7 +34,17 @@ cd /home/prow/go/src/github.com/kubeflow/pipelines/v2
...
+fi
+gcloud container clusters get-credentials "$TEST_CLUSTER" --region "$REGION" --project "$PROJECT"
+set +e
+kubectl port-forward svc/metadata-grpc-service 8080:8080 -n "$NAMESPACE" & PORT_FORWARD_PID=$!
+# wait for kubectl port forward
+sleep 10
 ${GO_CMD} test -v -cover ./...
+# kill kubectl port forward before exit
+echo "killing kubectl port forward before exit"
+kill "$PORT_FORWARD_PID"
```
Comment (Bobgy) [inline v2/test/presubmit-v2-go-test.sh:50]: Will the Pod get stuck in pending if we don't kill the port forward?

If necessary, then this should be in a trap (a syntax in bash like exit handler)

Comment (capri-xiyue) [reply v2/test/presubmit-v2-go-test.sh]: Done. (Previously I used `set +e` to make sure the kill command will get called. trap looks clearer.

**Comment by Bobgy** [comment]
/lgtm
/approve
Awesome, thanks for fixing this!

**Pull request merged**